CDCgov/prime-reportstream
# 9642 - Update Content and Link on API Programmer's Guide Card on Resources Page

Updated Content and Link on API Programmer's Guide Card on Resources Page

Test Steps:
1. Go to RS -> Resources
2. see new ReportStream API card
3. click on ReportStream API card and see new page

## Changes
- Resources Page block of "API Programmer's Guide" links to the main page of the new Web Version of the API Programmer's Guide
- Deleted legacy api programmers guide references.
- Users with old bookmarks to the old API Programmer's Guide page will be redirected to the new API Programmer's Guide page
- Renamed programmers-guide-api folder to reportstream-api
- Updated links that pointed to /resources/programmers-guide to /resources/api


## Checklist

### Testing
![Screenshot 2023-06-08 at 3 49 09 PM](https://github.com/CDCgov/prime-reportstream/assets/102491809/7086fd8a-2226-4dce-8735-8099391702dd)

![Screenshot 2023-06-08 at 3 49 47 PM](https://github.com/CDCgov/prime-reportstream/assets/102491809/9bca0147-bb92-4b83-a1da-e03c3915af36)

![Screenshot 2023-06-08 at 3 49 58 PM](https://github.com/CDCgov/prime-reportstream/assets/102491809/b08475fb-13e9-4cf5-a151-5729ad215db2)

## Linked Issues
- Fixes #9642
